### PR TITLE
State of the crates.io

### PR DESCRIPTION
**Pull request by sgrif**
Ashley asked me to write a blog post about what we've been doing on
crates.io the past few weeks. This is that blog post.

/cc @ashleygwilliams

### CODE REVIEW

File: _posts/2018-04-27-State-of-the-crates.io.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+The other endpoint which was extermely slow was "/crates". We started combing
+through the logs for these two endpoints to see if there were any patterns
+around the times that requests slowed down. At this point we realized we were
+being spammed by several crawlers at a rate we couldn't handle.
+
+We could have fixed this by upgrading our database server. We're on a pretty low
+tier of database server, and we have a lot of room to scale vertically on that
+front. However, not doing that forced us to solve several problems that we need
+to solve eventually. For example, we didn't have a published crawler policy ([we
+do now!](https://crates.io/policies)). This also forced us to introduce
```
Comment (sgrif) [inline _posts/2018-04-27-State-of-the-crates.io.md]: @ashleygwilliams We need to actually do this before this gets published. :)

File: _posts/2018-04-27-State-of-the-crates.io.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+When we first took over, things were not in great shape. The first and most
+pressing problem was that there was no monitoring set up. When there were major
+spikes in error responses, nobody was getting paged. The first thing we did was
+set up a pagerduty rotation.
+
+Inevitably when you add monitoring to something that didn't have it before, you
+learn that there are more problems than you realized. So we got paged... in the
+middle of the night... a lot.
+
+![Three incidents in the middle of the night](/images/2018-04-Crates.io/pages.png){:class="center"}
```
[the remark this answers is not in the document]
Comment (sgrif) [reply _posts/2018-04-27-State-of-the-crates.io.md]: On GitHub, not on the actual site.

**Comment by aturon** [comment]
cc @rust-lang/core

**Comment by sgrif** [comment]
This has gotten super stale, and is no longer relevant to publish.

**Pull request closed without merging**